exporter/awsemfexporter/metric_declaration.go

```diff
@@ -0,0 +1,171 @@
+// Copyright 2020, OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package awsemfexporter
+
+import (
+	"errors"
+	"regexp"
+	"sort"
+	"strings"
+
+	"go.opentelemetry.io/collector/consumer/pdata"
+	"go.uber.org/zap"
+)
+
+// MetricDeclaration characterizes a rule to be used to set dimensions for certain
+// incoming metrics, filtered by their metric names.
+type MetricDeclaration struct {
+	// Dimensions is a list of dimension sets (which are lists of dimension names) to be
+	// included in exported metrics. If the metric does not contain any of the specified
+	// dimensions, the metric would be dropped (will only show up in logs).
+	Dimensions [][]string `mapstructure:"dimensions"`
+	// MetricNameSelectors is a list of regex strings to be matched against metric names
+	// to determine which metrics should be included with this metric declaration rule.
+	MetricNameSelectors []string `mapstructure:"metric_name_selectors"`
+
+	// metricRegexList is a list of compiled regexes for metric name selectors.
+	metricRegexList []*regexp.Regexp
+}
+
+// Remove duplicated entries from dimension set.
+func dedupDimensionSet(dimensions []string) (deduped []string, hasDuplicate bool) {
+	seen := make(map[string]bool, len(dimensions))
+	for _, v := range dimensions {
+		seen[v] = true
+	}
+	hasDuplicate = (len(seen) < len(dimensions))
+	if !hasDuplicate {
+		deduped = dimensions
+		return
+	}
+	deduped = make([]string, len(seen))
+	idx := 0
+	for dim := range seen {
+		deduped[idx] = dim
+		idx++
+	}
+	return
+}
+
+// Init initializes the MetricDeclaration struct. Performs validation and compiles
+// regex strings. Dimensions are deduped and sorted.
+func (m *MetricDeclaration) Init(logger *zap.Logger) (err error) {
+	// Return error if no metric name selectors are defined
+	if len(m.MetricNameSelectors) == 0 {
+		return errors.New("invalid metric declaration: no metric name selectors defined")
+	}
+
+	// Filter out duplicate dimension sets and those with more than 10 elements
+	validDims := make([][]string, 0, len(m.Dimensions))
+	seen := make(map[string]bool, len(m.Dimensions))
+	for _, dimSet := range m.Dimensions {
+		concatenatedDims := strings.Join(dimSet, ",")
+		if len(dimSet) > 10 {
+			logger.Warn("Dropped dimension set: > 10 dimensions specified.", zap.String("dimensions", concatenatedDims))
```

Comment:
That's right, this only happens on startup. In this case, would it be ok to use `Warn` since it's only once on startup? I feel that this information is useful for the user to fix in case they don't enable `Debug` level logging.